cocos/2d/CCSprite.cpp

```diff
@@ -152,9 +154,11 @@ bool Sprite::initWithTexture(Texture2D *texture)
     CCASSERT(texture != nullptr, "Invalid texture for sprite");
 
     Rect rect = Rect::ZERO;
-    rect.size = texture->getContentSize();
+    if (texture) {
+        rect.size = texture->getContentSize();
+    }
 
-    return initWithTexture(texture, rect);
+    return initWithTexture(texture, rect, false);
```

Comment:
@zilongshanren  "initWithTexture(texture, rect);" will call "initWithTexture(texture, rect, false);", call "initWithTexture(texture, rect, false);" directly here is better for performance. I think the code is OK.